### PR TITLE
Port tests from FX/OldSrc/

### PR DESCRIPTION
**Pull request by ianhays**
Ports tests to cover the gaps between the tests in ToF/tests/FX/OldSrc/ and the tests in open.

going to need to rebase after merging #7562

### CODE REVIEW

File: src/System.Runtime/tests/System/DateTime.cs [hunk 1]
```diff
@@ -625,4 +642,15 @@ public object GetFormat(Type formatType)
             return typeof(IFormatProvider) == formatType ? this : null;
         }
     }
+
+    [Fact]
+    public static void InvalidDateTimeStyles()
+    {
+        CultureInfo culture = new CultureInfo("en-US");
```
Comment (ellismg) [inline src/System.Runtime/tests/System/DateTime.cs]: NIT: Is this local actually used?

**Comment by ellismg** [comment]
One minor nit, LGTM otherwise.

**Comment by mellinoe** [comment]
LGTM

File: src/System.Runtime/tests/System/DateTime.cs (hunk 1 above)
Comment (mellinoe) [inline src/System.Runtime/tests/System/DateTime.cs]: looks like it's accidentally re-created a few lines later.

File: src/System.Collections/tests/Generic/List/List.Generic.Tests.Remove.cs [hunk 2]
```diff
@@ -51,6 +51,12 @@ public void RemoveAll_DefaultElements(int count)
             Assert.Equal(expectedCount, removedCount);
         }
 
+        [Fact]
+        public void RemoveAll_NullMatchPredicate()
+        {
+            Assert.Throws<ArgumentNullException>(() => new List<T>().RemoveAll(null));
```
Comment (stephentoub) [inline src/System.Collections/tests/Generic/List/List.Generic.Tests.Remove.cs]: Nit: same comment as on the previous PR, but it'd be good to include parameter name information in Throws for ArgumentExceptions

File: src/System.IO/tests/StringReader/StringReader.CtorTests.cs [hunk 3]
```diff
@@ -51,7 +50,14 @@ public static void StringReaderWithGenericString()
...
                 Assert.Equal((int)str1[i], sr.Read());
             }
+        }
 
+        [Fact]
+        public static void ReadLine()
+        {
+            string str1 = "Hello\0\t\v   \\ World";
+            StringReader sr = new StringReader(str1);
+            Assert.Equal(str1, sr.ReadLine());
```
Comment (stephentoub) [inline src/System.IO/tests/StringReader/StringReader.CtorTests.cs]: What about the case where the input actually contains a newline?

**Comment by stephentoub** [comment]
A few nits, otherwise LGTM

File: src/System.Runtime/tests/System/DateTime.cs (hunk 1 above)
Comment (ianhays) [inline src/System.Runtime/tests/System/DateTime.cs]: Yup, I moved it into a below line but forgot to remove this one.

File: src/System.IO/tests/StringReader/StringReader.CtorTests.cs (hunk 3 above)
Comment (ianhays) [inline src/System.IO/tests/StringReader/StringReader.CtorTests.cs]: Good call. Added.

**Pull request merged**